toxics/latency.go

```diff
@@ -37,10 +37,14 @@ func (t *LatencyToxic) Pipe(stub *ToxicStub) {
 				return
 			}
 			sleep := t.delay() - time.Since(c.Timestamp)
+			start := time.Now()
 			select {
 			case <-time.After(sleep):
+				c.Timestamp = c.Timestamp.Add(sleep)
 				stub.Output <- c
 			case <-stub.Interrupt:
+				slept := time.Now().Sub(start)
```

Comment:
I'm not a big fan of using a timed value here, though in this instance I think it does work.
When the toxic is interrupted, it's a bit of an edge-case where we don't really care too much about the latency, just that it gets through.
It's up to you if you want to leave this here or not.